test/opexecuter.cpp

```diff
@@ -0,0 +1,359 @@
+                               const string op_name_prefix, int index,
+                               const DataType dt, NodeDef& node_def) {
+  string new_node_name = op_name_prefix + std::to_string(index);
+  node_def.set_name(new_node_name);
+  node_def.set_op(op_type);
+  SetAttrValue(dt, &((*(node_def.mutable_attr()))["T"]));
+  SetAttrValue(index, &((*(node_def.mutable_attr()))["index"]));
+}
+
+// Update data structures for book keeping
+// node_inedge_md : Map of
+//                  key : Node*
+//                  value : vector of pair{Node* src_incoming_edge, int
+//                  src_output_index}
+// node_outedge_md : Map of
+//                  key : Node*
+//                  value : vector of pair{Node* dst_outgoing_edge, int
+//                  dst_input_index}
+// node_outedges : Map of
+//                  key : Node*
+//                  value : vector of outgoing edges (Edge*)
+// test_op : update pointer to test_op pointer
+void OpExecuter::GetNodeData(Graph& graph, NodeMetaData& node_inedge_md,
+                             NodeMetaData& node_outedge_md,
+                             NodeOutEdges& node_outedges, Node** test_op) {
+  bool found_test_op = false;
+  for (const Edge* e : graph.edges()) {
+    if (!found_test_op) {
+      if (e->src()->IsOp() && (e->src()->type_string()) == test_op_type_) {
+        found_test_op = true;
+        *test_op = e->src();
+      }
+      if (e->dst()->IsOp() && (e->dst()->type_string()) == test_op_type_) {
+        found_test_op = true;
+        *test_op = e->dst();
+      }
+    }
+    NGRAPH_VLOG(5) << "Edge between, Src: " << e->src()->name()
+                   << " Src op index " << e->src_output()
+                   << " ,Dst: " << e->dst()->name() << " dst ip index "
+                   << e->dst_input();
+    // update src's outedge metadata
+    node_outedge_md[e->src()].push_back({e->dst(), e->dst_input()});
+    node_inedge_md[e->dst()].push_back({e->src(), e->src_output()});
+    node_outedges[e->src()].push_back(e);
+  }
+}
+
+// Validate that the graph has N allowed_nodes and 1 test_op_type node
+// Graph must look like this
+//
+// Const1     ConstN
+//   \    ...    /
+//    \         /
+//      Test_Op
+//
+// TODO check for vector allowed_nodes
+// when we allow other than "Const" node type as input
+// Make allowed_nodes const member of the class, use set
+void OpExecuter::ValidateGraph(const Graph& graph,
+                               const vector<string> allowed_nodes) {
+  NGRAPH_VLOG(5) << "Validate graph";
+  bool found_test_op = false;
+  for (Node* node : graph.nodes()) {
+    if (node->IsSource() || node->IsSink()) {
+      continue;
+    } else if (node->type_string() == test_op_type_) {
+      // only one node of type test_op
+      ASSERT_FALSE(found_test_op);
+      found_test_op = true;
+    } else {
+      ASSERT_TRUE(node->type_string() == allowed_nodes[0])
+          << "Found Not allowed Op: " << node->type_string();
+    }
+  }
+
```

Comment:
The asserts made on 'found_test_op' can probably be made in GetNodeData. In that case the ValidateGraph function can be removed

Reply:
That's correct, it can be moved there. The primary reason for implementing like this was to have clear responsibility for each function.  Since, the class (unit test) has well defined constraints (Only "Const" nodes + 1 "test_op_type" node) on what kind of graph it can accept, I decided to have a separate function for validating, and only then process a validated graph.